### PR TITLE
 The big format version 2

### PR DESCRIPTION
**Pull request by float3**

### CODE REVIEW

File: Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkEditor.cs [hunk 1]
```diff
@@ -5,35 +5,42 @@
...
                     EditorUtility.SetDirty(behaviour);
-                    
-                    if(PrefabUtility.IsPartOfPrefabInstance(behaviour))
+
+                    if (PrefabUtility.IsPartOfPrefabInstance(behaviour))
                     {
                         PrefabUtility.RecordPrefabInstancePropertyModifications(behaviour);
                     }
-                    
+                    PrefabUtility.RecordPrefabInstancePropertyModifications(behaviour);
```
Comment (pema99) [inline Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkEditor.cs:39]: ehh.. not just a format change here?

File: Packages/com.llealloo.audiolink/Runtime/Scripts/AudioLinkController.cs [hunk 2]
```diff
@@ -1,264 +1,257 @@
...
-                _X2 = PropertyToID("_X2");
-                _X3 = PropertyToID("_X3");
-                _Threshold0 = PropertyToID("_Threshold0");
-                _Threshold1 = PropertyToID("_Threshold1");
-                _Threshold2 = PropertyToID("_Threshold2");
-                _Threshold3 = PropertyToID("_Threshold3");
-            }
+        private void InitIDs()
+        {
+            _X0 = VRCShader.PropertyToID("_X0");
```
Comment (pema99) [inline Packages/com.llealloo.audiolink/Runtime/Scripts/AudioLinkController.cs:74]: doesn't this break standalone?

File: Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkEditor.cs (hunk 1 above)
Comment (float3) [reply Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkEditor.cs:39]: not sure how this happened, will resolve it

File: Packages/com.llealloo.audiolink/Runtime/Scripts/AudioLinkController.cs (hunk 2 above)
Comment (pema99) [reply Packages/com.llealloo.audiolink/Runtime/Scripts/AudioLinkController.cs:74]: oh, the whole script is `#if UDONSHARP` now ?

File: Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkEditor.cs [hunk 3]
```diff
@@ -5,35 +5,40 @@
...
         {
-            UdonSharpBehaviour[] allUdonSharpBehaviours = FindObjectsOfType<UdonSharpBehaviour>();
-            foreach (var behaviour in allUdonSharpBehaviours)
+            if (behaviour.GetType().GetField("audioLink") != null)
             {
-                if (behaviour.GetType().GetField("audioLink") != null)
+                behaviour.GetType().GetField("audioLink").SetValue(behaviour, target);
+                EditorUtility.SetDirty(behaviour);
+
+                if (PrefabUtility.IsPartOfPrefabInstance(behaviour))
```
Comment (pema99) [inline Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkEditor.cs:30]: you still made a change here. is this intentional?

Comment (pema99) [reply Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkEditor.cs:30]: image

Comment (float3) [reply Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkEditor.cs:30]: oops, it's resolved this time

File: Packages/com.llealloo.audiolink/Runtime/Scripts/AudioLinkController.cs (hunk 2 above)
Comment (float3) [reply Packages/com.llealloo.audiolink/Runtime/Scripts/AudioLinkController.cs:74]: I'm fine with keeping it multiplatform but you previously said that the controller is currently only for Udon, looking through the code I don't see anything vrc specific

**Review by pema99 — APPROVED** [review]

**Pull request merged**